server/controllers/ArticleController.js

```diff
@@ -0,0 +1,200 @@
+   * @param {Object} req the request body
+   * @param {Object} res the response body
+   * @param {function} next a call to the next function
+   * @returns {Object} the response body
+   */
+  static getAllArticle(req, res, next) {
+    Article.findAndCountAll({
+      include: [{
+        model: User,
+        as: 'author',
+        attributes: [
+          'username', 'bio', 'image',
+        ],
+      }],
+      attributes: [
+        'slug', 'title', 'description',
+        'body', 'createdAt', 'updatedAt',
+      ],
+      order: [['createdAt', 'DESC']],
+    })
+      .then((articles) => {
+        if (!articles.count) {
+          return res.status(404).json({
+            status: 'fail',
+            message: 'No Articles found',
+          });
+        }
+        return res.status(200).json({
+          articles: articles.rows,
+          articlesCount: articles.count,
+        });
+      })
+      .catch(next);
+  }
+
+  /**
+   * Method updates an article in the database,
+   * It returns a json response of the article updated
+   * It handles the PUT /api/articles/:slug.
+   * @param {Object} req the request body
+   * @param {Object} res the response body
+   * @param {function} next a call to the next function
+   * @returns {Object} the response body
+   */
+  static updateArticle(req, res, next) {
+    const {
+      title, description, body,
+    } = req.body.article;
+    const authorId = req.body.decoded.id;
+    const fieldsArr = [];
+
+    if (title) fieldsArr.push('title');
+    if (description) fieldsArr.push('description');
+    if (body) fieldsArr.push('body');
+
+    Article.update(
+      {
+        title, description, body,
+      },
+      {
+        where: { slug: req.params.slug, authorId },
+        returning: true,
+        fields: fieldsArr,
+      },
+    )
+      .then(([updated, article]) => {
+        if (!updated) {
+          return res.status(404).json({
+            status: 'fail',
+            message: 'Article not found. Cannot Update',
+          });
+        }
+        return res.status(200).json({
+          article,
+          message: 'Article Updated',
+        });
+      })
+      .catch(next);
+  }
+
+  /**
+   * Method deletes an article from the database,
+   * It returns a json response of the article deleted
+   * It handles the DELETE /api/articles/:slug.
+   * @param {Object} req the request body
+   * @param {Object} res the response body
+   * @param {function} next a call to the next function
+   * @returns {Object} the response body
+   */
+  static deleteArticle(req, res, next) {
+    const authorId = req.body.decoded.id;
+    Article.destroy({ where: { slug: req.params.slug, authorId } })
+      .then((success) => {
+        if (!success) {
+          return res.status(404).json({
+            status: 'fail',
+            message: 'Unable to delete Article',
+          });
+        }
+        return res.status(200).json({
+          message: 'Article Deleted',
```

Comment:
Add `status`:`success` to the response that is being sent

Reply:
done